examples/list-datatype.kl

```diff
@@ -0,0 +1,22 @@
+#lang "n-ary-app.kl"
+
+(import "defun.kl")
+
+(datatype (List A)
+  (nil)
+  (:: A (List A)))
```

Comment:
does using `::` instead of `cons` make sense if it can't be used infix?

Reply:
We use prefix + all the time in Lisp-land. Seems analogous.

But really, I've just used syntax from Pie when in doubt. I'm not attached to this name.